### DISCUSSION TITLE
Any workaround to implement look-around regex in tree-sitter?

### DISCUSSION THREAD

**Discussion by bookofproofs** [tree-sitter/tree-sitter#1189] [Q&A] [answered]
I have an EBNF grammar in which one rule matches any character until the string ":end" occurs. I implement this rule using the regular expression 

`/((?!:end).)*/` 

When generating grammar.js, I get the following error in tree-sitter:

`Error processing rule ExtensionContent
Details:
  regex parse error:
    ((?!:end).)*
     ^^^
error: look-around, including look-ahead and look-behind, is not supported`

Is there a work-around enabling me to create such a rule using tree-sitter?

**Comment by ahelwer** [comment, marked as answer]
Could you use `seq(/.*/, ':end')`?

**Reply by bookofproofs** [reply]
thank you, that helps!
